src/NewChain/ChainVerifier.ts

```diff
@@ -0,0 +1,414 @@
+    ) === -1
+    ) {
+      throw new Error('ContractsBin: Mismatch of Anchor BIN!!!');
+    }
+
+    const deployedOriginAnchorOrganizationBin = await this.originWeb3.eth.getCode(
+      this.contractAddresses.origin.anchorOrganizationAddress,
+    );
+    if (this.abiBinProvider.getBIN('Organization').toLowerCase().indexOf(
+      deployedOriginAnchorOrganizationBin.toLowerCase().slice(2),
+    ) === -1
+    ) {
+      throw new Error('ContractsBin: Mismatch of Organization BIN!!!');
+    }
+
+    const deployedAuxiliaryAnchorOrganizationBin = await this.originWeb3.eth.getCode(
+      this.contractAddresses.auxiliary.anchorOrganizationAddress,
+    );
+    if (this.abiBinProvider.getBIN('Organization').toLowerCase().indexOf(
+      deployedAuxiliaryAnchorOrganizationBin.toLowerCase().slice(2),
+    ) === -1
+    ) {
+      throw new Error('ContractsBin: Mismatch of Organization BIN!!!');
+    }
+
+    const deployedOriginOrganizationBin = await this.originWeb3.eth.getCode(
+      this.contractAddresses.origin.ostGatewayOrganizationAddress,
+    );
+    if (this.abiBinProvider.getBIN('Organization').toLowerCase().indexOf(
+      deployedOriginOrganizationBin.toLowerCase().slice(2),
+    ) === -1
+    ) {
+      throw new Error('ContractsBin: Mismatch of Organization BIN!!!');
+    }
+
+    const deployedAuxiliaryOrganizationBin = await this.auxiliaryWeb3.eth.getCode(
+      this.contractAddresses.auxiliary.ostCoGatewayOrganizationAddress,
+    );
+    if (this.abiBinProvider.getBIN('Organization').toLowerCase().indexOf(
+      deployedAuxiliaryOrganizationBin.slice(2).toLowerCase(),
+    ) === -1
+    ) {
+      throw new Error('ContractsBin: Mismatch of Organization BIN!!!');
+    }
+
+    const deployedOstPrimeBin = await this.auxiliaryWeb3.eth.getCode(
+      this.contractAddresses.auxiliary.ostPrimeAddress,
+    );
+    if (this.abiBinProvider.getBIN('OSTPrime').toLowerCase().indexOf(
+      deployedOstPrimeBin.slice(2).toLowerCase(),
+    ) === -1
+    ) {
+      throw new Error('ContractsBin: Mismatch of OSTPrime BIN!!!');
+    }
+    Logger.info('Successfully completed contracts BIN verification!!!');
+  }
+
+  /**
+   * This method does verification of Gateway storage data.
+   *
+   * @returns {Promise<void>}
+   */
+  private async verifyGateway(): Promise<void> {
+    const gatewayInstance = this.mosaicContract.EIP20Gateway(
+      this.contractAddresses.origin.ostEIP20GatewayAddress,
+    );
+
+    const isActivated = await gatewayInstance.methods.activated().call();
+    if (isActivated !== true) {
+      throw new Error('Gateway: It should be activated!!!');
+    }
+
+    const valueToken = await gatewayInstance.methods.token().call();
+    if (this.originWeb3.utils.toChecksumAddress(valueToken)
+        !== this.originWeb3.utils.toChecksumAddress(
+          this.mosaicConfig.originChain.contractAddresses.simpleTokenAddress,
+        )) {
+      throw new Error('Gateway: Invalid valueToken address!!!');
```

Comment:
We should also log the errors with expected and actual addresses. 
(For all such instance of error throughout the file.)